src/main/java/com/iexec/sms/secret/compute/AppComputeSecretController.java

```diff
@@ -97,7 +117,19 @@ public AppComputeSecretController(AuthorizationService authorizationService,
 
     @RequestMapping(method = RequestMethod.HEAD, path = "/apps/{appAddress}/secrets/{secretIndex}")
     public ResponseEntity<ApiResponseBody<String>> isAppDeveloperAppComputeSecretPresent(@PathVariable String appAddress,
-                                                                                     @PathVariable String secretIndex) {
+                                                                                         @PathVariable String secretIndex) {
+        try {
```

Comment:
It looks we can refactor this check in a method in order to be reused in `addAppDeveloperAppComputeSecret` & `isAppDeveloperAppComputeSecretPresent`

Reply:
exaclty, but I decided to go as straight as possible or it won't be possible to bring this story to a close.
We can create an issue and improve code later.

Reply:
This should be improved when enabling multiple application secret developers.
I've done some of the ground work, but going further should not be in the scope of the current story.

Reply:
see 9b2680e
By the way, the lines in the `try` block can be deduplicated, it will be harder for the lines in the `catch` block.

Reply:
finally, fixed in d0033ed